### PR TITLE
fix(iot-device): Add checks to verify that client property update response has version

### PR DESCRIPTION
**Pull request by abhipsaMisra**
For a twin update operation, not all responses have the request Id and version no. 
Eg. we know that request Id is only relevant for MQTT based operations.

Based on #2095 , it looks like module client operations with IoT Edge might be missing the version no. in the response.

This PR adds a check to see if the request Id and version are a part of the service returned response, and if yes, only then returns it to the device app.

### CODE REVIEW

File: iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs
Comment (drwill-ms) [inline iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs:1028]: You could inline all the calls to TryGetValue, rather than creating boolean return values you later reference.

**Review by drwill-ms — APPROVED** [review]

**Review by vinagesh — APPROVED** [review]

File: iothub/device/src/ClientPropertiesUpdateResponse.cs
Comment (abhipsaMisra) [inline iothub/device/src/ClientPropertiesUpdateResponse.cs:26]: compiler cannot resolve `<see cref="ClientProperties.ReportedFromClient.Version"/>`

File: iothub/device/src/ClientPropertyCollection.cs
Comment (abhipsaMisra) [inline iothub/device/src/ClientPropertyCollection.cs:38]: `<inheritdoc path="/exception"` inherits all exceptions listed under the API in `cref`.

File: iothub/device/src/ClientPropertyCollection.cs
Comment (abhipsaMisra) [inline iothub/device/src/ClientPropertyCollection.cs:55]: This API needs to inherit 1 of the 2 exceptions listed under `AddRootProperty`, so mentioning it explicitly

File: iothub/device/src/ClientPropertyCollection.cs
Comment (abhipsaMisra) [inline iothub/device/src/ClientPropertyCollection.cs:51]: Xml doc generation tool isn't able to resolve the query string within the inheritdoc path, so we cannot use this pattern.

File: iothub/device/src/ClientPropertyCollection.cs
Comment (abhipsaMisra) [inline iothub/device/src/ClientPropertyCollection.cs:60]: Inherit both exceptions under `AddRootProperty` but overwrite the description for `ArgumentNullException` by specifying it explicitly in L73

File: iothub/device/src/ClientPropertyCollection.cs
Comment (abhipsaMisra) [inline iothub/device/src/ClientPropertyCollection.cs:91]: need to inherit only 1 exception under `AddRootProperty`, so we'll need to specify it explicitly instead.

File: iothub/device/src/ClientPropertyCollection.cs
Comment (abhipsaMisra) [inline iothub/device/src/ClientPropertyCollection.cs:101]: need to inherit only 1 exception under `AddRootProperty`, so we'll need to specify it explicitly instead.

File: iothub/device/src/ClientPropertyCollection.cs
Comment (abhipsaMisra) [inline iothub/device/src/ClientPropertyCollection.cs:132]: need to inherit only 1 exception under `AddRootProperty`, so we'll need to specify it explicitly instead.

**Comment by abhipsaMisra** [comment]
Waiting for confirmation from the Edge team on the wording of this comment, before merging this PR in: https://github.com/Azure/azure-iot-sdk-csharp/blob/3d145a111f0f8caedb4f7218760f07fb183f0e94/iothub/device/src/ClientPropertiesUpdateResponse.cs#L24-L26

**Review by jamdavi — APPROVED** [review]

**Comment by abhipsaMisra** [comment]
> Waiting for confirmation from the Edge team on the wording of this comment, before merging this PR in:
> 
> https://github.com/Azure/azure-iot-sdk-csharp/blob/3d145a111f0f8caedb4f7218760f07fb183f0e94/iothub/device/src/ClientPropertiesUpdateResponse.cs#L24-L26

Edge team signed-off on this wording.

**Pull request merged**
